lib/internal/modules/esm/loader.js

```diff
@@ -96,6 +100,59 @@ const {
 
 let emittedSpecifierResolutionWarning = false;
 
+function nextHookFactory(chain, meta, validate) {
+  // First, prepare the current
+  const { hookName } = meta;
+  const {
+    fn: hook,
+    url: hookFilePath,
+  } = chain[meta.hookIndex];
+
+  const nextHookName = `next${
+    StringPrototypeToUpperCase(hookName[0]) +
+    StringPrototypeSlice(hookName, 1)
+  }`;
```

Comment:
This is to generate `nextResolve` or `nextLoad`?

Reply:
Yes. I'm not crazy about it, but passing an extra param just for that seemed worse.

Reply:
Could we have tests to ensure the names are what we expect them to be?

Reply:
done: 69eb1b2

Reply:
Why is an extra param worse?